### PR TITLE
Throwing error message incase of lengthy payload

### PR DESCRIPTION
**Pull request by shreyavshetty**
Payload match exceeding a certain limit will be truncated and the changes are made to throw an error message for the same.

### CODE REVIEW

**Comment by migueldemoura** [comment]
Builds are failing, can you run `cargo fmt`?

File: src/main.rs [hunk 1]
```diff
@@ -247,6 +252,22 @@ mod tests {
             str::from_utf8(&out.stdout).unwrap()
         );
 
+       let encrypted_matched_data_file_truncated = temp_dir.child("encrypted_matched_data_truncated.txt");
```
Comment (migueldemoura) [inline src/main.rs:255]: Should we move this to its own test?

File: src/main.rs [hunk 2]
```diff
@@ -118,6 +118,11 @@ fn run(options: Options) -> Result<(), String> {
                 fs::read_to_string(command.matched_data_filename)
                     .map_err(|_| "Failed to read matched data from file")?
             };
+
+            if matched_data_base64 == "truncated" {
```
Comment (migueldemoura) [inline src/main.rs:122]: nit: Should `truncated` be a const?

**Review by migueldemoura — CHANGES_REQUESTED** [review]

**Review by migueldemoura — APPROVED** [review]

**Comment by migueldemoura** [comment]
Can you sign your commits, @shreyavshetty (and merge the 1st and 2nd ones as the truncation bit, and rename the 3rd one to "Update dependencies" or something similar)?

**Comment by shreyavshetty** [comment]
> Can you sign your commits, @shreyavshetty (and merge the 1st and 2nd ones as the truncation bit, and rename the 3rd one to "Update dependencies" or something similar)?

Squashed to one, and signed

**Pull request merged**